hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/wal/FSHLog.java

```diff
@@ -1120,8 +1120,18 @@ public void shutdown() throws IOException {
           this.disruptor.shutdown();
         }
       }
-      // With disruptor down, this is safe to let go.
-      if (this.appendExecutor !=  null) this.appendExecutor.shutdown();
+
+      try {
+        // With disruptor down, this is safe to let go.
+        if (this.appendExecutor != null) {
+          this.appendExecutor.shutdown();
+          // Wait for 60 seconds to complete currently running tasks.
+          this.appendExecutor.awaitTermination(60, TimeUnit.SECONDS);
+        }
+      } catch (InterruptedException e) {
+        LOG.warn("Exception while waiting to shutdown executor ", e);
```

Comment:
Wondering if we should sleep for 5-10s by default here.

Reply:
Can you propagate the interrupt for completeness.

> Wondering if we should sleep for 5-10s by default here.

Curious why?

Reply:
> Can you propagate the interrupt for completeness.

👍 

> > Wondering if we should sleep for 5-10s by default here.
> 
> Curious why?

Since awaiting on termination of Executor did not go as expected, we might want to wait a little more? But now I think we don't need to (I take it back, no explicit wait required) and we are anyways going to interrupt current thread. Instead of waiting, we might rather prefer performing `shutdownNow()` which would interrupt running tasks? (if Disruptor consumer responds to interruption). WDYT?

Reply:
@shahrs87 I think you missed the comment on resetting the interrupt flag.

@virajjasani I don't think we need to shutdownNow() again because the JVM is shutting down anyway and that interrupts all the running threads (if any that exist at that point). If we are in that situation, we are already in a bad state, don't think a proper shutdownNow() would help much.

Reply:
> @shahrs87 I think you missed the comment on resetting the interrupt flag.

@bharathv You mean calling Thread.currentThread().interrupt(); in catch block ?

Reply:
Right.

Reply:
Done !